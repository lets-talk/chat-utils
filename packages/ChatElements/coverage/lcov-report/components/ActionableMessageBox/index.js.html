<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/ActionableMessageBox/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/ActionableMessageBox</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
&nbsp;
import './index.scss';
&nbsp;
const ActionableMessageBox = (props) =&gt; {
  const { data, onClickAction } = props;
  const { actions } = data;
&nbsp;
  return (
    &lt;ul className={classNames('letstalk-actionable-mbox')}&gt;
      {actions.map((action, index) =&gt;
        (
          &lt;li
            key={action.id}
            className="letstalk-actionable-item"
            role="button"
            onClick={onClickAction &amp;&amp; (<span class="fstat-no" title="function not covered" ><span class="branch-1 cbranch-no" title="branch not covered" >(e</span>) =&gt; <span class="cstat-no" title="statement not covered" >onClickAction(action, index, e))</span></span>}
          &gt;
            {action.name}
          &lt;/li&gt;
        ))}
    &lt;/ul&gt;
  );
};
&nbsp;
ActionableMessageBox.defaultProps = {
  onClickAction: null,
};
&nbsp;
ActionableMessageBox.propTypes = {
  data: PropTypes.object.isRequired,
  onClickAction: PropTypes.func,
};
&nbsp;
export default ActionableMessageBox;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 24 2017 01:32:25 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
