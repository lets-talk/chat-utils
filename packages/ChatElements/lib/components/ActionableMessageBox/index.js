import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import FaCloudDownload from 'react-icons/lib/fa/cloud-download';

import './index.scss';

const ActionableMessageBox = (props) => {
  const { data, onClickAction } = props;
  const { actions } = data;

  return (
    <ul className={classNames('letstalk-actionable-mbox')}>
      {actions.map((action, index) =>
        (
          <li
            key={action.id}
            className="letstalk-actionable-item"
            role="button"
            onClick={onClickAction && ((e) => onClickAction(action, index, e))}
          >
            {action.name}
          </li>
        ))}
    </ul>
  );
};

ActionableMessageBox.defaultProps = {
  data: {},
  onClickAction: null,
};

ActionableMessageBox.propTypes = {
  data: PropTypes.object,
  onClickAction: PropTypes.func,
};

export default ActionableMessageBox;
