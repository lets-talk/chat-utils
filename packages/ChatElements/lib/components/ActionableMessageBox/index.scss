@import './config.scss';

.letstalk-actionable-mbox {
  display: block;
  cursor: pointer;
  padding: 0;
  margin: 0;
  border-radius: 5px;
  background-color: $color-white;
  font-size: $font-size-caption;
  font-weight: $font-weight-semi-bold;

  li.letstalk-actionable-item {
    list-style: none;
    padding: 15px;
    @include themify($themes) {
      color: themed('color-background-primary');
    }
  }

  li.letstalk-actionable-item:hover {
    @include themify($themes) {
      background-color: rgba(themed('color-background-primary'), .05);
    }
  }

  li.letstalk-actionable-item:first-child {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
  }

  li.letstalk-actionable-item:last-child {
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
  }

  li.letstalk-actionable-item:not(:last-child) {
    @include themify($themes) {
      border-bottom: 1px solid themed('color-divider');
    }
  }

}
