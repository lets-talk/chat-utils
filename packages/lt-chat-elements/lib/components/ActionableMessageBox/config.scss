@import "../../common/styles/variables";
@import "../../common/styles/themes/themify";

$input-border-color: rgb(136, 196, 63);
