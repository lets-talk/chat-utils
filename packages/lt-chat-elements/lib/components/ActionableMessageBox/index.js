import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ActionList = styled.ul`
  display: block;
  cursor: pointer;
  padding: 0;
  margin: 0;
  border-radius: 5px;
  background-color: ${(props) => props.theme.palette.common.white};
  font-size: ${(props) => props.theme.typography.classes.caption.bold.accent.fontSize};
  font-weight: ${(props) => props.theme.typography.classes.caption.bold.accent.fontWeight};
`;

const ActionItem = styled.li`
  list-style: none;
  padding: 15px;
  color: ${(props) => props.theme.palette.action.active};

  &:hover {
    background-color: ${(props) => props.theme.palette.action.hover};
  }
  &:first-child {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
  }
  &:last-child {
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
  }
  &:not(:last-child) {
    border-bottom: 1px solid ${(props) => props.theme.palette.divider};
  }
`;

const ActionableMessageBox = (props) => {
  const { data, onClickAction } = props;
  const { actions } = data;

  return (
    <ActionList className={props.className}>
      {actions.map((action, index) =>
        (
          <ActionItem
            key={action.id}
            role="button"
            onClick={onClickAction && ((e) => onClickAction(action, index, e))}
          >
            <span>{action.name}</span>
          </ActionItem>
        ))}
    </ActionList>
  );
};

ActionableMessageBox.defaultProps = {
  onClickAction: null,
  className: 'LT-ActionableMessageBox-Container',
};

ActionableMessageBox.propTypes = {
  /**
   * Array of element to display as Action items [{ id: '1', name: 'MyAction'}]
   */
  data: PropTypes.object.isRequired,
  /**
   * Extra className to style the component
   */
  className: PropTypes.string,
  /**
   * Callback function called when user clicks on an action Item
   * The callback has the following signature:
   * callback (action, index, e)
   */
  onClickAction: PropTypes.func,
};

export default ActionableMessageBox;
