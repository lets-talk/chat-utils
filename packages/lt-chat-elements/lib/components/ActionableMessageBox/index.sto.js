import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, select, text } from '@storybook/addon-knobs';

// Component to show on storybook
import ActionableMessageBox from '.';
import { WrapWithTheme } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';

const stories = storiesOf('ActionableMessageBox', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('ActionableMessageBox'));

const data = {
  actions: [
    {
      id: 1,
      name: 'Emergencia Bancaria',
    },
    {
      id: 2,
      name: 'Tarjeta de crédito',
    },
    {
      id: 3,
      name: 'Crédito Personal',
    },
    {
      id: 4,
      name: 'Otro',
    },
  ],
};

stories.addWithInfo(
  'ActionableMessageBox',
  'This is the basic usage with the ActionableMessageBox',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: '10px' }}>
      <WrapWithTheme
        themeName={select('Theme', { default: 'DefaultTheme', light: 'LightTheme', dark: 'DarkTheme' }, 'default')}
      >
        <ActionableMessageBox
          onClickAction={action('onClick')}
          data={data}
        />
      </WrapWithTheme>
    </div>
  )
);
