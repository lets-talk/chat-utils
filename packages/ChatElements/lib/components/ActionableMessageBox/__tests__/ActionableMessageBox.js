import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ActionableMessageBox from '../index';

describe('ActionableMessageBox component', () => {
  const data = {
    actions: [
      {
        id: 1,
        name: 'Emergencia Bancaria',
      },
      {
        id: 2,
        name: 'Tarjeta de crédito',
      },
      {
        id: 3,
        name: 'Crédito Personal',
      },
      {
        id: 4,
        name: 'Otro',
      },
    ],
  };

  it('should render without data', () => {
    const component = shallow(<ActionableMessageBox data={data} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data', () => {
    const component = shallow(<ActionableMessageBox data={data} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
