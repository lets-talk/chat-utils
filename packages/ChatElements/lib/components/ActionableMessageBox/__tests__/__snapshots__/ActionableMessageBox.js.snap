// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionableMessageBox component should render with data 1`] = `
<ul
  className="letstalk-actionable-mbox"
>
  <li
    className="letstalk-actionable-item"
    key="1"
    onClick={null}
    role="button"
  >
    Emergencia Bancaria
  </li>
  <li
    className="letstalk-actionable-item"
    key="2"
    onClick={null}
    role="button"
  >
    Tarjeta de crédito
  </li>
  <li
    className="letstalk-actionable-item"
    key="3"
    onClick={null}
    role="button"
  >
    Crédito Personal
  </li>
  <li
    className="letstalk-actionable-item"
    key="4"
    onClick={null}
    role="button"
  >
    Otro
  </li>
</ul>
`;

exports[`ActionableMessageBox component should render without data 1`] = `
<ul
  className="letstalk-actionable-mbox"
>
  <li
    className="letstalk-actionable-item"
    key="1"
    onClick={null}
    role="button"
  >
    Emergencia Bancaria
  </li>
  <li
    className="letstalk-actionable-item"
    key="2"
    onClick={null}
    role="button"
  >
    Tarjeta de crédito
  </li>
  <li
    className="letstalk-actionable-item"
    key="3"
    onClick={null}
    role="button"
  >
    Crédito Personal
  </li>
  <li
    className="letstalk-actionable-item"
    key="4"
    onClick={null}
    role="button"
  >
    Otro
  </li>
</ul>
`;
